hoomd3/classic/butane/spectral_abf/butane_SpectralABF.py

```diff
@@ -0,0 +1,321 @@
+    dihedral.params["HCCC"] = dict(k1=0.0, k2=0.0, k3=0.3, k4=0.0)
+    dihedral.params["HCCH"] = dict(k1=0.0, k2=0.0, k3=0.3, k4=0.0)
+
+    lj_special_pairs = hoomd.md.special_pair.LJ()
+    lj_special_pairs.params["CCCC"] = dict(epsilon=0.07, sigma=3.55)
+    lj_special_pairs.params["HCCH"] = dict(epsilon=0.03, sigma=2.42)
+    lj_special_pairs.params["HCCC"] = dict(epsilon=sqrt(0.07 * 0.03), sigma=sqrt(3.55 * 2.42))
+    lj_special_pairs.r_cut["CCCC"] = 12.0
+    lj_special_pairs.r_cut["HCCC"] = 12.0
+    lj_special_pairs.r_cut["HCCH"] = 12.0
+    coulomb_special_pairs = hoomd.md.special_pair.Coulomb()
+    coulomb_special_pairs.params["CCCC"] = dict(alpha=0.5)
+    coulomb_special_pairs.params["HCCC"] = dict(alpha=0.5)
+    coulomb_special_pairs.params["HCCH"] = dict(alpha=0.5)
+    coulomb_special_pairs.r_cut["HCCH"] = 12.0
+    coulomb_special_pairs.r_cut["CCCC"] = 12.0
+    coulomb_special_pairs.r_cut["HCCC"] = 12.0
+    integrator = hoomd.md.Integrator(dt=dt)
+    integrator.forces.append(lj)
+    integrator.forces.append(coulomb[0])
+    integrator.forces.append(coulomb[1])
+    integrator.forces.append(harmonic)
+    integrator.forces.append(angle)
+    integrator.forces.append(dihedral)
+    integrator.forces.append(lj_special_pairs)
+    integrator.forces.append(coulomb_special_pairs)
+    nvt = hoomd.md.methods.Langevin(filter=hoomd.filter.All(), kT=kT)
+    integrator.methods.append(nvt)
+    sim.operations.integrator = integrator
+
+    return sim
+
+
+# %%
+def get_args(argv):
+    available_args = [
+        ("fit-freq", "f", int, 100, "Fitting frequency"),
+        ("time-steps", "t", int, 5e5, "Number of simulation steps"),
+    ]
+    parser = argparse.ArgumentParser(description="Example script to run SpectralABF")
+    for name, short, T, val, doc in available_args:
+        parser.add_argument("--" + name, "-" + short, type=T, default=T(val), help=doc)
+
+    return parser.parse_args(argv)
+
+
+# %%
+def main(argv=[]):
+    args = get_args(argv)
+
+    cvs = [DihedralAngle([0, 4, 7, 10])]
+    grid = Grid(lower=(-pi,), upper=(pi,), shape=(64,), periodic=True)
+    method = SpectralABF(cvs, grid,fit_freq=args.fit_freq)
+
+    timesteps = args.time_steps
+    tic = time.perf_counter()
+    run_result = pysages.run(method, generate_context, timesteps)
+    toc = time.perf_counter()
+    print(f"Completed the simulation in {toc - tic:0.4f} seconds.")
+
+    # extract free energy from result
+    result = pysages.analyze(run_result)
+    mesh = result["mesh"]
+    fes_fn = result["fes_fn"]
+    A = fes_fn(mesh)
+    A = A.max() - A
+
+    # plot the free energy
+    fig, ax = plt.subplots()
+
+    ax.set_xlabel(r"Dihedral Angle, $\xi$")
+    ax.set_ylabel(r"$A(\xi)$")
+
+    ax.plot(mesh, A)
+    plt.gca()
+    fig.savefig("butane-fe.png")
+
+    # write free energy to file
+    dih_vs_A = np.stack([mesh[:, 0], A[:, 0]], axis=1)
+    np.savetxt("butane-fe.dat", dih_vs_A, header='"dih", "Free energy"')
```

Comment:
From .dat to pickle (https://github.com/SSAGESLabs/PySAGES-examples/pull/14/files)

Reply:
Pickle was added in https://github.com/SSAGESLabs/PySAGES-examples/pull/15/commits/385c0aed6121b764767d117aeea5f6d8056d275e